local/php/envs_test.go

```diff
@@ -182,6 +182,138 @@ func (s *PHPFPMSuite) TestGenerateEnv(c *C) {
 				"SCRIPT_NAME":     "/index.php",
 			},
 		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory",
+			expected: map[string]string{
+				"PATH_INFO":       "",
+				"REQUEST_URI":     "/subdirectory",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/",
+			expected: map[string]string{
+				"PATH_INFO":       "/",
+				"REQUEST_URI":     "/subdirectory/",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/unknown.php",
+			expected: map[string]string{
+				"PATH_INFO":       "/unknown.php",
+				"REQUEST_URI":     "/subdirectory/unknown.php",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/unknown.php/",
+			expected: map[string]string{
+				"PATH_INFO":       "/unknown.php/",
+				"REQUEST_URI":     "/subdirectory/unknown.php/",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/index.php/foo",
+			expected: map[string]string{
+				"PATH_INFO":       "/foo",
+				"REQUEST_URI":     "/subdirectory/index.php/foo",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/subdirectory/",
+			expected: map[string]string{
+				"PATH_INFO":       "/",
+				"REQUEST_URI":     "/subdirectory/subdirectory/",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "///subdirectory",
+			expected: map[string]string{
+				"PATH_INFO":       "///subdirectory",
+				"REQUEST_URI":     "///subdirectory",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/index.php",
+				"SCRIPT_NAME":     "/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory///subdirectory//foo/",
+			expected: map[string]string{
+				"PATH_INFO":       "///subdirectory//foo/",
+				"REQUEST_URI":     "/subdirectory///subdirectory//foo/",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
```

Comment:
TBH, I did that to preserve the same functionality as currently you have with default PHP file which was that you would get uri like //something. Said that, the non default file had completely different behaviour either way, so either way it will not be consistent. If you say that's good, I am good with it too.